docs/migratation-guide-v1.0.rst

```diff
@@ -0,0 +1,311 @@
+
+Differences
+^^^^^^^^^^^
+
+-  ``Connection`` constructor accepts single object parameter.
+-  ``Connection.run()`` follows Node.js convention of error first `error
+   first
+   callbacks <https://nodejs.org/api/errors.html#errors_error_first_callbacks>`__.
+
+Migrating from ``iConn`` to ``Connection`` Constructor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code:: js
+
+   // using idb transport
+
+   // const conn = new iConn("*LOCAL", "myuser", "mypassword");
+
+   const conn = new Connection({
+     transport: 'idb',
+     transportOptions: { database: '*LOCAL', username: 'myuser', password: 'mypassword' }
+   });
+
+   // using rest transport
+
+   // const restConfig = { host: 'myhost', port: 80, path: '/' }
+   // const conn = new iConn('*LOCAL', 'myuser', 'mypassword', restConfig);
+
+   const conn = new Connection({
+     transport: 'rest',
+     transportOptions: {
+       database: '*LOCAL',
+       username: 'myuser',
+       password: 'mypassword',
+       url: 'http://myhost:80/',
+     }
+   });
+
+Migrating from ``iConn.run()`` to ``Connection.run()``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Create an instance of Connection with ``returnError`` set to false.
+   This is a compatabilty option to behave like ``iConn.run()`` and
+   return the xml output as the first parameter of the run callback.
+
+.. code:: js
+
+   // const conn = new iConn("*LOCAL", "myuser", "mypassword");
+
+   const conn = new Connection({
+     transport: 'idb',
+     returnError: false,
+     transportOptions: { database: '*LOCAL', username: 'myuser', password: 'mypassword' }
+   });
+
+   conn.add(...)
+
+   conn.run((xmlOutput) => {
+       ...
+   })
+
+2. Test your application still works as expected using this instance of
+   ``Connection``.
+
+3. Update ``Connection.run()`` callbacks to expect an error as the first
+   parameter.
+
+.. code:: js
+
+   conn.run((error, xmlOutput) => {
+       if (error) { throw error; }
+   });
+
+4. Remove ``returnError`` property from the ``Connection`` constructor.
+   The default behavior is to return error first callbacks.
+
+.. code:: js
+
+   const conn = new Connection({
+     transport: 'idb',
+     transportOptions: { database: '*LOCAL', username: 'myuser', password: 'mypassword' }
+   });
+
+iPgm
+----
+
+``iPgm`` was replaced by the ``ProgramCall`` and will be removed in
+``v2.x``.
+
+Differences
+^^^^^^^^^^^
+
+-  Data and data structures and are now defined as objects.
+-  ``ProgramCall.addParam()`` now accepts a single object parameter.
+-  ``ProgramCall.addReturn()`` now accepts a single object parameter,
+-  ``ProgramCall.addReturn()`` now supports DS as return type.
+
+Migrating from ``iPgm.addParam()`` to ``ProgramCall.addParam()``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Parameter and data options are passed with the object parameter. Ensure
+you specify the data type odefaulting to use ``1024a`` is deprecated.
```

Comment:
typo on defaulting. Also seems to need some sort of grammar/punctuation - maybe a dash or a semicolon between type and defaulting? not sure what was meant here.